### ISSUE TITLE
status bars?

### ISSUE THREAD

**Issue by FennMacon** [sidevalve/sidevalve#14]
do you think we should have status bars like health/mana/stamina?

health would be health obviously but i was thinking mana could be how high you are, you get mana by smoking and it goes down by 5 points every time you go to a new destination (maybe set the mana drain per location?)

**Comment by amonks** [comment]
idk. There are definitely possible games that could benefit from a feature like that, but there are just as many games that wouldn't. 

I'd be ok with adding status stuff as long as it wasn't sbrbn-vntr specific: someone making a different game needs to be able to call it something other than `highness` in their game.js.

**Comment by FennMacon** [comment]
true, I was thinking about that too, something like giving the option for status bars and the ability to name them and say what locations/actions depleted them

**Comment by amonks** [comment]
if it was location based as u say, maybe something like this:
```javascript
game = {
	"healthIsCalled": "highness"

	"player": {
		startingHealth = 50
	}

	"places": {
		"lounge": {
			"healthChange": +5,
			"healthNeededToEnter: 12
		},
		"lamePlace": {
			"healthChange": -2,
		}
	}

	"objects": {
		"weedpipe": {
			"healthBonusWhenYouPickItUp": 100
		}
	}
}
```

note how the player can name it health and it's pretty extensible as to how a game uses it

**Comment by amonks** [comment]
or maybe track multiple statuses?

game = {
	"healthLikeThings": {
		"highness",
		"health"
	}

	"player": {
		startingScores: {
			 "highness": 0,
			 "health": 100
		}
	}

	"places": {
		"lounge": {
			"impacts": {
				"health": +5,
				"highness": +10
			},
			"requires": {
				"highness": 50
			}
		},
	}

	"objects": {
		"weedpipe": {
			"impacts": {
				"highness": 100
			}
		}
	}
}
